### ISSUE TITLE
PycodestyleBear and LineLengthBear print redundant messages

### ISSUE THREAD

**Issue by Techievena** [coala/coala-bears#1394]
Result with PycodestyleBear:
![image](https://cloud.githubusercontent.com/assets/17270966/22578452/e5e791bc-e9ee-11e6-850c-09c96adf6b10.png)
Result with LineLengthBear:
![image](https://cloud.githubusercontent.com/assets/17270966/22578483/203b2b1c-e9ef-11e6-9c90-e55d2e10d060.png)

**Comment by Makman2** [comment]
Yeah `PycodestyleBear` does also line-length-checking. We can't avoid this, and imo this is okay.

@Techievena some reason you really want to change it?

**Comment by Techievena** [comment]
When running coala for the first time in large projects these redundant messages are really frustating. ( Got frustated today when running our coafile on apertium_lint and that is even a very small project) This redundancy is OK for small fixes but when if we are extend coala for project wide support we must remove these redundancies imo. :smile:

**Comment by Makman2** [comment]
In this case you can just remove the `SpaceConsistencyBear`^^ Then you don't have redundant messages.

Imo this gets way better streamlined as soon as we have aspects (https://github.com/coala/cEPs/blob/master/cEP-0005.md).

**Comment by Techievena** [comment]
Sorry for late reply. Wasn't around here for last 2-3 days. But SpaceConsistencyBear is really an useful bear when we consider analysis of new projects imo. We shouldn't remove that. By the way great proposals at 
https://github.com/coala/cEPs/blob/master/cEP-0005.md . :smiley: You should close this issue then. Looking forward to be a part of the team implementing this proposal.

**Comment by Makman2** [comment]
Yes but I mean don't use both bears at once for now ;)

Let's keep this issue anyway open, maybe we find something we can easily integrate in our current system. As you're right, we are duplicating results here^^

**Comment by damngamerz** [comment]
@Techievena I suggest we can simply '--ignore' that E501 Error in PycodestyleBear.So as to remove its redundancy with LinelengthBear.what say?I will be happy to work on this :+1:

**Comment by Techievena** [comment]
@damngamerz See this https://github.com/coala/coala-bears/blob/master/.coafile#L9 Also we can ignore certain lines using --ignore not any type of error.

**Comment by damngamerz** [comment]
@Techievena Im talking about PycodestyleBear we can ignore any type of error either creating a specific config, or adding ```--ignore``` tag in CLI https://pycodestyle.readthedocs.io/en/latest/intro.html#configuration

**Comment by Techievena** [comment]
Wait for @Makman2 's suggestion 😃

**Comment by Makman2** [comment]
Problem is `Pycodestyle` should work with the default settings when used out-of-the-box imo. If someone only specifies the `Pycodestyle` bear, I don't want to disable common-sense default settings.

But yes, using `--ignore` is a workaround for the coafile to avoid result duplication :+1: The bear already supports the setting `pycodestyle_ignore`.

**Comment by damngamerz** [comment]
@Makman2 @Techievena Then i think this issue can be closed :+1: .But for this kind of situation one necessarily should have the specific knowledge about that bear.Cant we do something about this ?or maybe we can hard code to ignore E501 specifically for this issue and then add to documentation(which will not be a good option but can be one of the solution). What say any other suggestions?

**Comment by Makman2** [comment]
> or maybe we can hard code to ignore E501in pycodestylebear

that's what I want to avoid ;)

**Comment by aptrishu** [comment]
we need some tweaks with Results being printed. and I agree with @Makman2 , to remove duplication we should not change the default behaviour of pycodestyleBear.

**Comment by aptrishu** [comment]
Though I wonder how good that kind of tweak would workout with aspects. (logically)

**Comment by Makman2** [comment]
With aspects we won't have this problem any more, if one wants to check line-length, there will be only a single check in the whole repo for it. If we would use `PycodestyleBear` for other aspects, we would really hardcode the E501 ignore into it.
